File: app/src/debug/java/org/mifos/mobile/api/ApiInterceptor.java
package org.mifos.mobile.api;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Request.Builder;
import okhttp3.Response;

public class ApiInterceptor implements Interceptor {

    public static final String HEADER_TENANT = "X-Tenant-Identifier";
    private String headerTenant;

    public ApiInterceptor(String headerTenant) {
        this.headerTenant = headerTenant;
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        Request chainRequest = chain.request();
        Builder builder = chainRequest.newBuilder()
                .header(HEADER_TENANT, headerTenant);

        Request request = builder.build();
        return chain.proceed(request);
    }
}

File: app/src/main/java/org/mifos/mobile/data/ApiInterceptor.kt
package org.mifos.mobile.data

import okhttp3.Interceptor
import okhttp3.Request
import okhttp3.Response
import java.io.IOException

class ApiInterceptor constructor(headerTenant: String): Interceptor {
    val HEADER_TENANT = "Platform-TenantId"
    lateinit var headerTenant: String
    init {
        this.headerTenant=headerTenant
    }

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response? {
        val chainRequest: Request = chain.request()
        val builder: Request.Builder = chainRequest.newBuilder()
                .header(HEADER_TENANT,headerTenant)
        val request: Request = builder.build()
        return chain.proceed(request)
    }
}